dask_jobqueue/remote_slurm.py

```diff
@@ -0,0 +1,115 @@
+from contextlib import contextmanager
+
+import requests
+
+from .slurm import SLURMCluster
+from .core import cluster_parameters, job_parameters
+from .slurm import SLURMJob, logger
+
+
+class RemoteSLURMJob(SLURMJob):
+    def __init__(self, *args, **kwargs):
+        self.api_url = kwargs.pop("api_url")
+        self.__class__.api_url = self.api_url
+        super().__init__(*args, **kwargs)
+
+    @contextmanager
+    def job_file(self):
+        """
+        Override the job_file and pass back the job_script. job_file() is used by Job.start and
+         gives output directly to SLURMJob._submit_job().
+        """
+        yield self.job_script()
+
+    async def _submit_job(self, script):
+        # https://slurm.schedmd.com/rest_api.html#slurmctldSubmitJob
+        try:
+            response = requests.post(
+                f"{self.api_url}jobs/submit", json={"script": script}
+            )
```

Comment:
Given that this call is being made within an `async` method it will block the event loop.

It would be better to use `aiohttp` here.

Reply:
Thank you for the recommendation. :) I've implemented aiohttp now, however the test is failing with;

```
E               RuntimeError: Task <Task pending coro=<SpecCluster._correct_state_internal() running at distributed/deploy/spec.py:326> cb=[<TaskWakeupMethWrapper object at 0x1126262e8>()]> got Future <Future pending> attached to a different loop
```

I suppose this is because the loop SpecCluster does not get passed to the job and thus the job cannot add tasks to the loop. Jobs also do not get passed the loop currently.

I have very limited knowledge of asyncio, rarely work with it... Would you be able to assist here @jacobtomlinson?

Reply:
They shouldn't need to be passed the loop. Just implement async methods.

Reply:
Got it to work :v: